src/compiler/checker.ts

```diff
@@ -13443,7 +13443,7 @@ namespace ts {
         }
 
         function getTypeOfFirstParameterOfSignature(signature: Signature) {
-            return getTypeAtPosition(signature, 0);
+            return getTypeWithFacts(getTypeAtPosition(signature, 0), TypeFacts.NEUndefined);

```

Comment:
well. i went back and forth on this, and then did not see a reason why not... 
so we want to get `T`

`then<U>(success?: (value: T) => U,...): IPromise<U>`

the first takes care of the optionality of `succes?`, the second would remove the `undefined` if `value` was optional. the question is what is the promised type for this:

`then<U>(success?: (value?: T) => U,...): IPromise<U>`

is it `T` or `T|undefined` ?